packages/broker/src/plugins/operator/MaintainTopologyHelper.ts

```diff
@@ -125,29 +124,28 @@ export class MaintainTopologyHelper extends EventEmitter<MaintainTopologyHelperE
         }
         const parseItems = (response: any) => {
             if (!response.operator) {
-                this.logger.error(`Operator ${this.address.toLowerCase()} not found in TheGraph`)
+                logger.error('Unable to find operator in The Graph', { operatorContractAddress: this.operatorContractAddress })
                 return []
             }
             return response.operator.stakes
         }
         this.theGraphClient.updateRequiredBlockNumber(requiredBlockNumber)
-        const queryResult = this.theGraphClient.queryEntities<any>(createQuery, parseItems)
+        const queryResult = this.theGraphClient.queryEntities<any>(createQuery, parseItems) // TODO: add type
 
         for await (const stake of queryResult) {
-            if (stake.sponsorship.stream && stake.sponsorship.stream.id
-                && this.streamIdOfSponsorship.get(stake.sponsorship.id) !== stake.sponsorship.stream.id) {
-                const streamId = stake.sponsorship.stream.id
-                this.streamIdOfSponsorship.set(stake.sponsorship.id, stake.sponsorship.stream.id)
+            const sponsorshipId = stake.sponsorship?.id
+            const streamId = toStreamIDSafe(stake.sponsorship?.stream?.id)
+            if (streamId !== undefined && sponsorshipId !== undefined && this.streamIdOfSponsorship.get(sponsorshipId) !== streamId) {
```

Comment:
Are all null checks needed? Maybe the GraphQL schema guarantees that we always have e.g. `stream` field inside a `Sponsorship` entity?

Reply:
I'll add a `TODO` on this as the point of this PR is to just improve the existing code somewhat and not attempt to solve everything.